### PR TITLE
Rescue - Add buttons to choose between password and SSH key authentication

### PR DESCRIPTION
**Pull request by Dampen59**
Ref : MANAGER-12571

| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)

## Description

Added Password and SSH Key authentication choices in the dedicated servers netboot wizard
The available authentication methods depends on the selected rescue kernel.

## Related

MANAGER-12571

### Translations
- [x] CMT-9664

### CODE REVIEW

File: packages/manager/modules/bm-server-components/src/netboot/translations/Messages_fr_FR.json [hunk 1]
```diff
@@ -38,5 +39,10 @@
   "server_configuration_netboot_confirm": "Valider",
   "server_configuration_netboot_review": "Résumé",
   "server_configuration_netboot_save_in_progress": "Mise à jour...",
-  "server_configuration_netboot_invalid": "Invalide"
+  "server_configuration_netboot_invalid": "Invalide",
+  "server_configuration_netboot_auth_method": "Méthode d'authentification :",
+  "server_configuration_netboot_auth_method_sshkey": "Authentification par clé SSH",
+  "server_configuration_netboot_auth_method_password": "Authentification par mot de passe",
+  "server_configuration_netboot_sshkey_input": "Votre clé SSH Publique :",
+  "server_configuration_netboot_sshkey_input_placeholder": "Insérez une valeur de type sshPubKey. Exemple : ssh-rsa AAAAXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX== my-public-key"
```
Comment (JayBeeDe) [inline packages/manager/modules/bm-server-components/src/netboot/translations/Messages_fr_FR.json:47]: @Dampen59 `ssh-rsa AAAAXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX== my-public-key` should be in packages/manager/modules/bm-server-components/src/netboot/constants.js
And translation should contains a variable, so  `ssh-rsa AAAAXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX== my-public-key` is passed as argument to the translation
```suggestion
  "server_configuration_netboot_sshkey_input_placeholder": "Insérez une valeur de type sshPubKey. Exemple : {{t0}}"
```

**Comment by JayBeeDe** [comment]
Squash all your commits together (data from separated commits will be lost anyway when your PR will be merged)

**Comment by JayBeeDe** [comment]
image
When ssh key is invalid, native error message is not shown.

Invalid error message can be shown only in the following case:
Empty data error is raised, then invalid ssh key is inserted
image

**Comment by qpavy** [comment]
Be carefull, I see you retrieve translation but QA isn't done yet. We have to check for QA to be ok before ask for translation

**Review by qpavy — APPROVED** [review]

**Comment by JayBeeDe** [comment]
> Be carefull, I see you retrieve translation but QA isn't done yet. We have to check for QA to be ok before ask for translation

I did the short-circuit to go faster, this is my mistake

File: packages/manager/modules/bm-server-components/src/netboot/template.html [hunk 2]
```diff
@@ -88,6 +90,51 @@
...
+                    <div class="row mb-2">
+                        <div
+                            class="col-md-4"
+                            data-ng-repeat="(key, data) in $ctrl.rescueAuthMethods"
+                        >
+                            <oui-select-picker
+                                name="{{:: data }}"
+                                model="$ctrl.currentAuthMethod"
+                                label="{{:: ('server_configuration_netboot_auth_method_'+ data) | translate }}"
+                                values="[data]"
```
Comment (qpavy) [inline packages/manager/modules/bm-server-components/src/netboot/template.html:102]:
```suggestion
                                data-name="{{:: data }}"
                                data-model="$ctrl.currentAuthMethod"
                                data-label="{{:: ('server_configuration_netboot_auth_method_'+ data) | translate }}"
                                data-values="[data]"
```

File: packages/manager/modules/bm-server-components/src/netboot/template.html [hunk 3]
```diff
@@ -88,6 +90,51 @@
...
+                            >
+                            </oui-select-picker>
+                        </div>
+                    </div>
+
+                    <div data-ng-if="$ctrl.currentAuthMethod === $ctrl.SSHKEY">
+                        <oui-field
+                            label="{{:: 'server_configuration_netboot_sshkey_input' | translate}}"
```
Comment (qpavy) [inline packages/manager/modules/bm-server-components/src/netboot/template.html:110]:
```suggestion
                            data-label="{{:: 'server_configuration_netboot_sshkey_input' | translate}}"
```

**Review by qpavy — APPROVED** [review]

**Review by JayBeeDe — APPROVED** [review]

**Review by qpavy — APPROVED** [review]

**Pull request merged**